Create param documentation (#202)
kitbagjs/router: docs/core-concepts/route-params.md

With this getter defined, now our route will only match if the param matches our rules above.

> As a `ParamGetter`, the value in `route.params` is still writable, but the set will assume `value.toString()` is sufficient. Alternatively if you use the `ParamGetSet` syntax, you can provide the same validation on value set as well.

Reviewer: ```suggestion
If you only supply a getter for your param it will assume that `value.toString()` is sufficient when setting the value. Alternatively you can supply both a `get` and a `set` method and provide validation on the value when setting as well.
```

])
```

> If you're using `createParam`, you can also assign a default value with the 2nd argument.

Reviewer: ```suggestion
You can also assign a default value with the 2nd argument of `createParam`.
```
